brickstrap-image.sh

```
@@ -0,0 +1,208 @@
+        return 1
+    else
+        eval "$(brp_get_image_type_driver "$1")" "$BRP_CUR_IMG" || \
+            brp_image_err_msg "$?" "$1" "$BRP_CUR_IMG"
+    fi
+}
+
+#
+# Look up the driver function for a given image type name.
+# $1: the name of the partitioning/imaging scheme implemented in the disk image
+#
+function brp_get_image_type_driver()
+{
+    [ $# -eq 1 -a -n "$1" ] && case "$1" in
+        default)
+            echo -n brp_image_default_driver
+        ;;
+        *)
+            eval echo -n "\$BRP_IMG_DRV_REGISTRY_$1"
+        ;;
+    esac
+}
+
+#
+# Look up the file type extension for a given image type name.
+# $1: the name of the partitioning/imaging scheme implemented in the disk image
+#
+function brp_get_image_type_extension()
+{
+    [ $# -eq 1 -a -n "$1" ] && case "$1" in
+        default)
+            echo -n img
+        ;;
+        *)
+            eval echo -n "\$BRP_IMG_EXT_REGISTRY_$1"
+        ;;
+    esac
+}
+
+#
+# Look up the validator function to check image configuration parameters.
+# $1: the name of the partitioning/imaging scheme implemented in the disk image
+#
+function brp_get_image_cfg_validator()
+{
+    [ $# -eq 1 -a -n "$1" ] && case "$1" in
+        default)
+            echo -n brp_validate_default_image_config
+        ;;
+        *)
+            eval echo -n "\$BRP_IMG_CFG_VALIDATOR_$1"
+        ;;
+    esac
+}
+
+#
+# Register a custom driver with brickstrap for a custom image type.
+# $1: the name of the partitioning/imaging scheme implemented in the disk image
+# $2: name of the function which will create the image (type), taking the image
+#     file path as argument.
+# $3: the image file type extension, without leading dot.
+# $4: optional: a validator function to check parameters intended for the image
+#     driver function in the project configuration.
+#
+function br_register_image_type()
+{
+    [ $# -ge 3 -a -n "$1" -a -n "$2" -a -n "$3" ] && \
```

Since this is a public function, I would still rather see an error message here rather than failing silently if the parameters are bad.